### PR TITLE
Fix for FileNotFoundError for plugins

### PR DESCRIPTION
**Pull request by PrajwalM2212**
FileNotFoundError occurs when running moban with plugins. This error occurred in both my string.Template and genshi plugin.

### CODE REVIEW

**Comment by PrajwalM2212** [comment]
@chfw You have to remove python 2.6 from build.

**Comment by PrajwalM2212** [comment]
https://github.com/moremoban/moban/pull/95 superseded by this PR

**Comment by chfw** [comment]
hei, I had a look. 

The problem is that there has been a change in development process. The change is:

1. any changes should raise PR to dev branch only
2. any release will be made when a merge to master is done.

I have been away for a while so I did not realize that I have made such a change.

**Comment by PrajwalM2212** [comment]
@chfw Oh ok...no problem.

File: moban/engine.py [hunk 1]
```diff
@@ -129,17 +128,29 @@ def _render_with_finding_data_first(self, data_file_index):
                 self.__file_count += 1
 
     def _apply_template(self, template, data, output):
+        temp_dir = ''
```
Comment (chfw) [inline moban/engine.py]: please put this part of code outside as a function, and then write a unit test on the your new function.

File: moban/engine.py [hunk 2]
```diff
@@ -213,8 +224,8 @@ def verify_the_existence_of_directories(dirs):
         if os.path.exists(directory):
             continue
         should_I_ignore = (
-            constants.DEFAULT_CONFIGURATION_DIRNAME in directory
-            or constants.DEFAULT_TEMPLATE_DIRNAME in directory
+                constants.DEFAULT_CONFIGURATION_DIRNAME in directory
```
Comment (chfw) [inline moban/engine.py]: extra indentation is introduced

**Comment by chfw** [comment]
And I recommend to run "black" tool on top of your change file to fix coding style

**Comment by chfw** [comment]
dev branch has already remove python 2.6 so your next commit shall be OK. https://github.com/moremoban/moban/blob/dev/.travis.yml

**Comment by chfw** [comment]
Likely you are more into python 3. For python 2, it is a legacy matintenance and will surport it till 2020. I don't want to drop python 2.7 users yet.

So please change from unittest.mock to just from mock, the test shall pass.

**Comment by PrajwalM2212** [comment]
@chfw I did the refactoring you asked me to do. Also added test. The travis is also passing.

**Comment by chfw** [comment]
Hei, thanks. I saw it and withdrew my comments.

**Pull request merged**